packages/checkbox/src/Checkbox.ts

```diff
@@ -126,11 +126,7 @@ export class Checkbox extends SizedMixin(CheckboxBase) {
             }
         }
         if (changes.has('indeterminate')) {
-            if (this.indeterminate) {
-                this.inputElement.setAttribute('aria-checked', 'mixed');
-            } else {
-                this.inputElement.removeAttribute('aria-checked');
-            }
+            this.inputElement.indeterminate = this.indeterminate;
```

Comment:
I think this needs to be a bit more robust. Otherwise, an indeterminate checkbox will announce the checked/unchecked state on click, even while `this.indeterminate` remains `true`.

Here is a suggestion:

```ts
        if (changes.has('indeterminate') || this.indeterminate) {
            this.inputElement.indeterminate = this.indeterminate;
            if (this.indeterminate) {
                this.checked = false;
            } else {
                this.inputElement.checked = this.checked;
            }
        }
```

Reply:
See https://github.com/adobe/spectrum-web-components/pull/3268

Reply:
Agree that there's something more here, but the interaction with `this.checked` begs some more thought on other things that might not have been covered previously.

1. Is an `indeterminate` Checkbox also `checked === false`?
2. When an `indeterminate` Checkbox is clicked, should it move to a state and remove `indeterminate`?
3. When an `indeterminate` checkbox is clicked, if it changes state, should it _always_ go to `checked === true` or should it go to the opposite of the last state before it became `interderminate`?
4. The suggestion above outlines a `checked === true` Checkbox that is moved to `indeterminate` and then later leaves `indeterminate` (without use interaction) will end up as `checked === false`. Is that correct?

Reply:
Looking more closely at a native `<input type="checkbox" />` I don't think we want `indeterminate` to effect `checked`. https://codepen.io/Westbrook/pen/poxMVWE

Here we see that whatever the value of `checked`, clicking and `indeterminate` `<input type="checkbox" />` will switch the value of `checked` and set `indeterminate = false`.

Is there some non-platform advice beyond that (there often is, but this is clearly a depth on this pattern deeper that I've researched before) to point to doing anything differently here?

Reply:
For now, I've added some tests and expanded the story for `indeterminate` to back us on up this stance. We can chat more next week, if there's more info we should take into account.

Reply:
If take the stand that the value of the input's checked state takes precedence, we should remove the indeterminate style when `checked === true`.

Reply:
Value over indeterminate seems to be the order of operations for native Checkboxes, so it stands that we should do the same. I've included the removal of `indeterminate` on any change (whether to `true` or to `false`) of `checked`. Let me know if you think that covers the intersection of realities here enough to move forward in an accessible way.

Reply:
This seems fine now.

Reply:
Adding this makes the difference:
https://github.com/adobe/spectrum-web-components/pull/3267/files#diff-f726e9e9adb1c2e8f11bc0649f910bf51ec08b9fc4ab3e3e528a46a78e8f3fc6R108-R112

Reply:
Thanks for helping us clear that up!